### PR TITLE
chore: brought in @codecademy/eslint-config for ESLint 8

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Brings in https://github.com/Codecademy/client-modules/pull/4 for a refreshed linting config.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1691
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618c00f881faa90079a8d4e4--gamut-preview.netlify.app>https://618c00f881faa90079a8d4e4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618c00f881faa90079a8d4e4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618c069853410f00ac5d97cb--gamut-preview.netlify.app>https://618c069853410f00ac5d97cb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618c069853410f00ac5d97cb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618c1b2002d27f1a030f2d4b--gamut-preview.netlify.app>https://618c1b2002d27f1a030f2d4b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618c1b2002d27f1a030f2d4b>Deploy Logs</a></p>

File: .eslintrc.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ module.exports = {
           },
         ],
         'local-rules/gamut-import-paths': 'error',
+
+        // TODO: Removing in client-modules...
+        'react/prop-types': 'off',
+        'react/default-props-match-prop-types': 'off',
```
Comment (JoshuaKGoldberg) [inline .eslintrc.js:31]: Jake fixed publishing today. I can fix this up before release 😄 ... these rules are disabled in the newer versions on the branch.

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "yarn": "^1.21.1"
   },
   "scripts": {
+    "compile": "tsc",
```
Comment (JoshuaKGoldberg) [inline package.json:13]:
```suggestion
```

File: packages/gamut/src/Spinner/index.tsx [hunk 3]
```diff
@@ -4,12 +4,8 @@ export type SpinnerProps = SVGProps<SVGSVGElement> & {
   size?: number | string;
 };
 
-const defaultProps: SpinnerProps = {
-  size: 24,
-};
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Spinner/index.tsx:9]: I ended up disabling the rule that banned these, but figured I might as well inline the simpler ones...

File: typings/css.d.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 declare module '*.css' {
-  const styles: { [i: string]: any };
+  const styles: { [i: string]: string };
   export default styles;
 }
 
 declare module '*.scss' {
-  const styles: { [i: string]: any };
+  const styles: { [i: string]: string };
```
Comment (JoshuaKGoldberg) [inline typings/css.d.ts:7]: `no-unsafe-any` rules were complaining here.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.6bb587.0<br/>@codecademy/gamut-labs@28.0.8-alpha.6bb587.0<br/>@codecademy/gamut@40.2.1-alpha.6bb587.0<br/>@codecademy/monaco@0.5.10-alpha.6bb587.0<br/>@codecademy/styleguide@50.1.10-alpha.6bb587.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618c1e4353410f16625d9bed--gamut-preview.netlify.app>https://618c1e4353410f16625d9bed--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618c1e4353410f16625d9bed>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.23-alpha.9e27d8.0<br/>@codecademy/gamut-labs@28.1.3-alpha.9e27d8.0<br/>@codecademy/gamut@40.5.2-alpha.9e27d8.0<br/>@codecademy/monaco@0.5.18-alpha.9e27d8.0<br/>@codecademy/styleguide@50.4.3-alpha.9e27d8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61a6590e6d849745ce47ad36--gamut-preview.netlify.app>https://61a6590e6d849745ce47ad36--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61a6590e6d849745ce47ad36>Deploy Logs</a></p>

**Pull request merged**